Source/GoogleMapsCarbonFootprint.js

```diff
@@ -233,10 +270,51 @@ function convertDistance(distanceStr) {
  * Return:
  *   - the carbon footprint, in grams.
  */
-function computeFootprint(distance) {
-  var footprint = distance * carbonEmission;
-  console.log('Carbon Footprint for this route is: ' + footprint + ' grams');
-  return footprint;
+
+ /*This function converts the time string to integer Time values */
+function convertTime(timestr) {
+
+    var str_new = timestr.split(" ");
+    console.log(str_new);
+
+    if (str_new[1] == "min") {
+        var time = parseInt(str_new[0]) / 60;
+    } else if (str_new[1] == "h" && str_new[3] != "min") {
+        var time = parseInt(str_new[0]);
+
+    } else if (str_new[1] == "h" && str_new[3] == "min") {
+        var time = parseInt(str_new[0]) + parseInt(str_new[2]) / 60;
+    } else {
+        var time = -1;
+    }
+    console.log("fimnal time is" + time);
+    return time;
+}
+
+
+function computeFootprint(distance, time) {
+
+    /*CALCULATE AVERAGE SPEED AS PER GOOGLE MAPS*/
+    if (time == -1) { averageSpeed = 30; } else {
+        var averageSpeed = distance / time;
+    }
+    console.log("average" + averageSpeed);
+
+    /*User given carbon emmission are taken on the basis on the average speed*/
+    if (averageSpeed >= 0 && averageSpeed < 10) {
+        carbonEmission = carbonEmission1;
+    } else if (averageSpeed >= 10 && averageSpeed < 30) {
+        carbonEmission = carbonEmission1;
+    } else if (averageSpeed >= 30 && averageSpeed <= 60) {
+        carbonEmission = carbonEmission2;
+    } else if (averageSpeed >= 60 && averageSpeed < 90) {
```

Comment:
You could merge the last two cases, because you do the same thing in both of them.